internal/driver/fetch_test.go

```diff
@@ -206,6 +206,109 @@ func TestFetch(t *testing.T) {
 	}
 }
 
+func TestFetchWithBase(t *testing.T) {
+	const path = "testdata/"
+	type testcase struct {
+		desc          string
+		sources       []string
+		bases         []string
+		normalize     bool
+		expectSamples bool
+	}
+
+	testcases := []testcase{
+		{
+			"not normalized, base is same as source",
+			[]string{path + "cppbench.cpu"},
+			[]string{path + "cppbench.cpu"},
+			false,
+			false,
+		},
+		{
+			"not normalized, single source, multiple base (all profiles same)",
+			[]string{path + "cppbench.cpu"},
+			[]string{path + "cppbench.cpu",path + "cppbench.cpu"},
+			false,
+			true,
+		},
+		{
+			"not normalized, different base and source",
+			[]string{path + "cppbench.cpu"},
+			[]string{path + "go.crc32.cpu"},
+			false,
+			true,
+		},
+		{
+			"normalized, base is same as source",
+			[]string{path + "cppbench.cpu"},
+			[]string{path + "cppbench.cpu"},
+			true,
+			false,
+		},
+		{
+			"normalized, single source, multiple base (all profiles same)",
+			[]string{path + "cppbench.cpu"},
+			[]string{path + "cppbench.cpu",path + "cppbench.cpu"},
+			true,
+			false,
+		},
+		{
+			"normalized, different base and source",
+			[]string{path + "cppbench.cpu"},
+			[]string{path + "go.crc32.cpu"},
+			true,
+			true,
+		},
+	}
+
+	for _, tc := range testcases {
```

Comment:
Not sure if you want to change this now, but sometimes running test cases using https://golang.org/pkg/testing/#T.Run makes the test code more readable. It allows using t.Fatal when a check fails that the test shouldn't skip, instead of using t.Error() + continue. And will make defer() statement execute for each test case individually which is sometimes expected.